va/validation-authority.go

```diff
@@ -373,8 +373,12 @@ func (va *ValidationAuthorityImpl) validateHTTP01(ctx context.Context, identifie
 	}
 
 	if expectedKeyAuth != payload {
+		truncBody := body
+		if len(body) > 40 {
+			truncBody = body[:40]
```

Comment:
nit: since you are using a byte slice this can corrupt unicode code points. It would also be nice to have some indication the body has been truncated other than assuming all 40 character long messages have been.